src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -141,10 +142,16 @@ protected override void Dispose(bool disposing)
         {
             if (disposing)
             {
-                if (null != _sslContext)
+                SafeSslHandle sslContext = _sslContext;
+                if (null != sslContext)
                 {
-                    _sslContext.Dispose();
-                    _sslContext = null!;
+                    lock (sslContext)
+                    {
+                        _inputBuffer.Dispose();
+                        _outputBuffer.Dispose();
+                    }
+
+                    sslContext.Dispose();
```

Comment:
I'm not really familiar with this code base, but I'm still uneasy with this. The SSL Context is being used locked in many other methods in this class, except in this line here, where it is disposed.  Maybe it's the case of having an object with the specific purpose of using it for locking instead?

Reply:
I like the idea of separate lock objects. This also may not be needed when #32925 lands. 
(I would need to investigate OSX more but I think it may be possible with changes at SslStream level)